eks/terraform/jenkins-x/irsa.tf

```diff
@@ -26,7 +26,7 @@ data "aws_iam_policy_document" "tekton-bot-policy" {
 
 resource "aws_iam_policy" "tekton-bot" {
```

Comment:
what's about documenting the resources a bit better? I like the idea of grouping the terraform files into sections, providing links to relevant docs, etc. Have a look at https://github.com/hferentschik/jx-cloud-provisioners/blob/issue-7/gke/terraform/main.tf. WDYT?

Reply:
why 'jenkins-x'? I feel that could be confusing in terms of people thinking we install Jenkins X there. I was more thinking about _cluster_, _storage_, _vault_. At the moment I also have _sa_ (service account) which is a bit work in progress. Not sure whether this eventually makes sense. Looking also a bit at https://github.com/jenkins-x/terraform-google-jx.

Would be nice to align on this.

Reply:
My thinking behind this module was to create any resources needed directly by Jenkins X to make its features work:
* buckets
* Route53 hosted zone
* Kubernetes Namespaces
* IRSA

I don't want to create too many modules as it will actually have a detrimental effect on understandability of this whole thing.

Reply:
> I don't want to create too many modules as it will actually have a detrimental effect on understandability of this whole thing.

I am not so sure about that. I'll complete what I am up to and then we can compare.